doc/cpd/101.rst

```diff
@@ -11,117 +11,182 @@ Description
+By adding encryption to sensitive data we could mitigate access from those with direct
+access to data dumps and storage systems.
 
 
 Design
 ------
-The ``StorageService`` instances would house a GnuPG configuration and keys allowing
-them to encrypt and decrypt data. The instances would need to do one of the following:
+The ``StorageService`` would be updated to know what data would be backed through Custodia
+rather than the other storage handler(s).
+
+- Install and setup of a Custodia instance would be part of a Commissaire install and configuration.
+- Custodia would be configured to use an etcd backend.
+- Custodia would be configured to use unix socket communication.
+- Commissaire's Storage Service would be updated to store credentials and ssh keys via Custodia.
+
+
+::
+
+                     +-----------------+
+     Data Request    |                 |
+   +---------------->+ Storage Service |
+                     |                 | +---------------+
+                     +-------+---------+ |               |
+                             |           |   Custodia    |
+                             |           |               |
+                             |           +---------------+
+                             v                 ^
+                     +-------+-------+      Yes|
+                     |               |         |
+                     |    Secret?    +---------+
+                     |               |         |
+                     +---------------+       No|
+                                               |
+                                               |
+                                               v
+                                         +-----------------------+
+                                         |                       |
+                                         |   Storage Handler(s)  |
+                                         |                       |
+                                         +-----------------------+
 
-* all have the same key
-* each instance have a copy of other instances public keys
-* use the bus to request the encryption key if the local key store is empty
 
 
 Additional Libraries
 ~~~~~~~~~~~~~~~~~~~~
-To facilitate easy access to GnuPG, `python-gnupg <http://pythonhosted.org/gnupg/>`_ would
-need to be installed. This library utilizes ``gpg`` directly so it would need to be
-available as well.
+`Custodia <https://custodia.readthedocs.io/en/latest/>`_ would be a required subsystem.
+Custodia would be installed as part of Commissaire.
 
 
 StorageService Updates
 ~~~~~~~~~~~~~~~~~~~~~~
-The StorageService would need to handle decrypting the key back into a base64 encoded
-string before handing it back to the requesting service.
+The StorageService would need to to know when to use Custodia versus the configured
+``StorageHandler``(s). It would look at the ``_secrets`` attribute on the instance and,
+if set to ``True`` would use the secrets handler.
```

Comment:
This sentence was confusing at first since the `_secrets` attribute isn't defined until a few paragraphs below.